devices/FloatingBaseEstimatorDevice/src/FloatingBaseEstimatorDevice.cpp

```diff
@@ -0,0 +1,519 @@
+
+FloatingBaseEstimatorDevice::FloatingBaseEstimatorDevice(double period,
+                                                         yarp::os::ShouldUseSystemClock useSystemClock)
+        : yarp::os::PeriodicThread(period, useSystemClock)
+{
+}
+
+FloatingBaseEstimatorDevice::FloatingBaseEstimatorDevice()
+        : yarp::os::PeriodicThread(0.01, yarp::os::ShouldUseSystemClock::No)
+{
+}
+
+FloatingBaseEstimatorDevice::~FloatingBaseEstimatorDevice()
+{
+}
+
+bool FloatingBaseEstimatorDevice::open(yarp::os::Searchable& config)
+{
+    YarpUtilities::getElementFromSearchable(config, "robot", m_robot);
+    YarpUtilities::getElementFromSearchable(config, "port_prefix", m_portPrefix);
+    YarpUtilities::getElementFromSearchable(config, "base_link_imu", m_baseLinkImuName);
+    YarpUtilities::getElementFromSearchable(config, "left_foot_wrench", m_leftFootWrenchName);
+    YarpUtilities::getElementFromSearchable(config, "right_foot_wrench", m_rightFootWrenchName);
+    double devicePeriod{0.01};
+
+    if (YarpUtilities::getElementFromSearchable(config, "sampling_period_in_s", devicePeriod))
+    {
+        setPeriod(devicePeriod);
+    }
+
+    if (!setupRobotModel(config))
+    {
+        return false;
+    }
+
+    if (!setupRobotSensorBridge(config))
+    {
+        return false;
+    }
+
+    if (!setupFeetContactStateMachines(config))
+    {
+        return false;
+    }
+
+    if (!setupBaseEstimator(config))
+    {
+        return false;
+    }
+
+    if (!YarpUtilities::getElementFromSearchable(config, "publish_rostf", m_publishROSTF))
+     {
+         m_publishROSTF = false;
+     }
+
+     if (m_publishROSTF)
+     {
+         if (!loadTransformBroadcaster())
+         {
+             return false;
+         }
+     }
+
+    return true;
+}
+
+bool FloatingBaseEstimatorDevice::setupRobotModel(yarp::os::Searchable& config)
+{
+
+    std::string modelFileName;
+    std::vector<std::string> jointsList;
+    if (!YarpUtilities::getElementFromSearchable(config, "model_file", modelFileName))
+    {
+        yError() << "[FloatingBaseEstimatorDevice][setupRobotModel] Missing required parameter \"model_file\"";
+        return false;
+    }
+
+    if (!YarpUtilities::getVectorFromSearchable(config, "joint_list", jointsList))
+    {
+        yError() << "[FloatingBaseEstimatorDevice][setupRobotModel] Missing required parameter \"joint_list\"";
+        return false;
+    }
+
+    yarp::os::ResourceFinder& rf = yarp::os::ResourceFinder::getResourceFinderSingleton();
+    std::string modelFilePath{rf.findFileByName(modelFileName)};
+    yInfo() << "[FloatingBaseEstimatorDevice][setupRobotModel] Loading model from " << modelFilePath;
+
+    iDynTree::ModelLoader modelLoader;
+    if (!modelLoader.loadReducedModelFromFile(modelFilePath, jointsList))
+    {
+        yError() << "[FloatingBaseEstimatorDevice][setupRobotModel] Could not load robot model";
+        return false;
+    }
+
+    m_model = modelLoader.model().copy();
```

Comment:
That was actually required in MATLAB (because a reference is returned). 
This is not needed in C++. Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/130/commits/fe323bd315395e6d00aaee756b971823696aeb92.